#### TCGA survival analysis: PAAD & LUAD
#### Goal for this analysis is to show whether activation of KEAP1-NRF2 pathway 
#### correlates with poor survival

# Load libraries
library(survival)
library(survminer)
library(SummarizedExperiment)
library(TCGAbiolinks)

### preload gene sets data

# old version of PKD signature
PKD.old <- read.csv("OLD_SIGNATURE.csv") %>% 
  mutate(X=NULL, gs_name = "PKD.old") 
PKD.old.ls <- PKD.old$ensembl_gene_id

# new version of PKD signature

PKD.new <- read.csv("NEW_SIGNATURE.csv") %>% 
  mutate(X=NULL, gs_name = "PKD.new") 
PKD.new.ls <- PKD.new$ensembl_gene_id

# direct NRF2 targets signature
## "direct NRF2 targets" was generated by filtering PKD.new signature with genes overlapped with
## ChIP-seq analysis showing direct NRF2 interaction

direct.NRF2 <- read.csv("dir_NRF2_SIGNATURE.csv") %>% 
  mutate(X=NULL, gs_name = "direct_NRF2") 
direct.NRF2.ls <- direct.NRF2$ensembl_gene_id

# Option: Download TCGA gene expression data for lung cancer (LUAD) or pancreatic cancer (LUAD)
# # Function to download and prepare data for a given cancer type
# get_tcga_data <- function(cancer_type) {
#   # Query for gene expression (RNA-Seq) data
#   query <- GDCquery(
#     project = paste0("TCGA-", cancer_type),
#     data.category = "Transcriptome Profiling",
#     data.type = "Gene Expression Quantification",
#     workflow.type = "STAR - Counts"
#   )
#   
#   # Download and prepare data
#   GDCdownload(query)
#   data <- GDCprepare(query)
#   
#   return(data)
# }
# 
# # Download data for PAAD and LUAD
# paad_data <- get_tcga_data("PAAD")
# luad_data <- get_tcga_data("LUAD")

# Load pre-downloaded PAAD and LUAD data
paad_data <- read_rds("../15_TCGA/TCGA_PAAD.rds")
luad_data <- read_rds("../15_TCGA/TCGA_LUAD.rds")

# Option: Download TCGA-mutation data data (use LUAD as an example) 
# query_mut <- GDCquery(
#   project = "TCGA-LUAD",
#   data.category = "Simple Nucleotide Variation",
#   data.type     = "Masked Somatic Mutation",
#   workflow.type = "Aliquot Ensemble Somatic Variant Merging and Masking"
# )
# GDCdownload(query_mut)
# maf_data <- GDCprepare(query_mut)
# write_rds(maf_data, "LUAD_mutation_data.rds")


# load pre-downloaded LUAD data
maf_data <- readRDS("LUAD_mutation_data.rds")

# Filter for KRAS mutant LUAD patients
kras_mut <- maf_data[maf_data$Hugo_Symbol == "KRAS", ]
kras_patients <- unique(substr(kras_mut$Tumor_Sample_Barcode, 1, 12))

# Expression matrix extraction function
extract_expression <- function(se) {
  expr <- assay(se, "tpm_unstrand")
  expr_log2 <- log2(expr + 1)
  return(expr_log2)
}

# Extract and format expression data
paad_expr <- extract_expression(paad_data)
rownames(paad_expr) <- gsub("\\..*", "", rownames(paad_expr))
colnames(paad_expr) <- sub("^((TCGA-[A-Z0-9]+-[A-Z0-9]+)).*", "\\1", colnames(paad_expr))

luad_expr <- extract_expression(luad_data)
rownames(luad_expr) <- gsub("\\..*", "", rownames(luad_expr))
colnames(luad_expr) <- sub("^((TCGA-[A-Z0-9]+-[A-Z0-9]+)).*", "\\1", colnames(luad_expr))

luad_kras_expr <- luad_expr[, colnames(luad_expr) %in% kras_patients] # filter patients with KRAS mutation


# Gene set score function (mean expression of log2(TPM+1) for each genes)
calculate_gene_set_score <- function(expr_matrix, gene_set) {
  genes_present <- intersect(gene_set, rownames(expr_matrix))
  if(length(genes_present) == 0){
    stop("None of the genes in the set are present in the expression matrix.")
  }
  score <- colMeans(expr_matrix[genes_present, ], na.rm = TRUE)
  return(score)
}

# Calculate gene set scores
paad_score_NRF2 <- calculate_gene_set_score(paad_expr, direct.NRF2.ls)
paad_score_old <- calculate_gene_set_score(paad_expr, PKD.old.ls)
paad_score_new <- calculate_gene_set_score(paad_expr, PKD.new.ls)

luad_score_NRF2 <- calculate_gene_set_score(luad_expr, direct.NRF2.ls)
luad_score_old <- calculate_gene_set_score(luad_expr, PKD.old.ls)
luad_score_new <- calculate_gene_set_score(luad_expr, PKD.new.ls)

luad_kras_score_NRF2 <- calculate_gene_set_score(luad_kras_expr, direct.NRF2.ls)
luad_kras_score_old <- calculate_gene_set_score(luad_kras_expr, PKD.old.ls)
luad_kras_score_new <- calculate_gene_set_score(luad_kras_expr, PKD.new.ls)

# Load clinical data
paad_clinical <- read_rds("../15_TCGA/PAAD_clinical.rds")
luad_clinical <- read_rds("../15_TCGA/LUAD_clinical.rds")
luad_kras_clinical <- luad_clinical %>% filter(submitter_id %in% kras_patients)

##############################################################################################
# Prepare survival data (this function should take gene scores from three different gene sets)
##############################################################################################


# Function to prepare survival data
prepare_survival_data <- function(clinical_data, gene_score_A, gene_score_B, gene_score_C) {
  # Ensure that patient IDs match
  # Extract the barcode (TCGA-XX-XXXX) to match expression data
  # Assuming column names in expr_matrix correspond to patient barcodes
  # Adjust as necessary based on your data
  clinical_data <- clinical_data %>%
    mutate(barcode = substr(bcr_patient_barcode, 1, 12)) # Adjust if needed
  
  # Create a data frame with survival info and gene scores
  survival_data <- clinical_data %>%
    select(barcode, days_to_death, days_to_last_follow_up, vital_status) %>%
    mutate(
      survival_time = ifelse(vital_status == "Dead", days_to_death, days_to_last_follow_up),
      status = ifelse(vital_status == "Dead", 1, 0)
    ) %>%
    select(barcode, survival_time, status)
  
  # Merge with gene scores
  survival_data$score_A <- gene_score_A[match(survival_data$barcode, names(gene_score_A))]
  survival_data$score_B <- gene_score_B[match(survival_data$barcode, names(gene_score_B))]
  survival_data$score_C <- gene_score_C[match(survival_data$barcode, names(gene_score_C))]
  
  
  # Remove samples with missing data
  survival_data <- na.omit(survival_data)
  
  return(survival_data)
}

# Merge expression scores with clinical survival info
paad_surv <- prepare_survival_data(paad_clinical, paad_score_new, paad_score_old, paad_score_NRF2) %>%
  rename(PKD.new_score = score_A, PKD.old_score = score_B, direct.NRF2_score = score_C)

luad_surv <- prepare_survival_data(luad_clinical, luad_score_new, luad_score_old, luad_score_NRF2) %>%
  rename(PKD.new_score = score_A, PKD.old_score = score_B, direct.NRF2_score = score_C)

luad_kras_surv <- prepare_survival_data(luad_kras_clinical, luad_kras_score_new, luad_kras_score_old, luad_kras_score_NRF2) %>%
  rename(PKD.new_score = score_A, PKD.old_score = score_B, direct.NRF2_score = score_C)



##########################################################
# Survival analysis function
## run survival analysis and generate Kaplan-Meier plots
##########################################################

perform_survival_analysis <- function(surv_data, gene_set_name) {
  # Create a survival object
  surv_data <- surv_data %>%
    mutate(group = ifelse(.data[[gene_set_name]] > median(.data[[gene_set_name]], na.rm = TRUE), "High", "Low"))
  
  # Create a survival object
  surv_object <- Surv(time = surv_data$survival_time, event = surv_data$status)
  
  # Fit survival curves
  fit <- survfit(Surv(survival_time, status) ~ group, data = surv_data)
  
  # Perform Log-rank test
  surv_diff <- survdiff(Surv(survival_time, status) ~ group, data = surv_data)
  p_value <- 1 - pchisq(surv_diff$chisq, length(surv_diff$n) - 1)
  
  # Plot survival curves
  plot <- ggsurvplot(
    fit,
    data = surv_data,
    pval = TRUE,
    pval.method = TRUE,
    title = paste("Survival Analysis for", gene_set_name),
    legend.title = gene_set_name,
    legend.labs = c("High", "Low"),
    xlab = "Time (days)",
    ylab = "Survival Probability",
    risk.table = TRUE
  )
  
  return(list(fit = fit, p_value = p_value, plot = plot))
}

# Run survival analyses
paad_analysis_A <- perform_survival_analysis(paad_surv, "PKD.new_score")
paad_analysis_B <- perform_survival_analysis(paad_surv, "PKD.old_score")
paad_analysis_C <- perform_survival_analysis(paad_surv, "direct.NRF2_score")

luad_analysis_A <- perform_survival_analysis(luad_surv, "PKD.new_score")
luad_analysis_B <- perform_survival_analysis(luad_surv, "PKD.old_score")
luad_analysis_C <- perform_survival_analysis(luad_surv, "direct.NRF2_score")

luad_kras_analysis_A <- perform_survival_analysis(luad_kras_surv, "PKD.new_score")
luad_kras_analysis_B <- perform_survival_analysis(luad_kras_surv, "PKD.old_score")
luad_kras_analysis_C <- perform_survival_analysis(luad_kras_surv, "direct.NRF2_score")